### PR TITLE
created addon column and updated email

### PR DESCRIPTION
**Pull request by qiuandyq**

### CODE REVIEW

**Comment by qiuandyq** [comment]
@kevqiu review please

File: prisma/migrations/20210907043419_change_name_field/migration.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*
```
Comment (kevqiu) [inline prisma/migrations/20210907043419_change_name_field/migration.sql:1]: how come 2 migrations

File: prisma/schema.prisma [hunk 2]
```diff
@@ -24,4 +24,13 @@ model Slot {
   service         String?
   instagramHandle String?
   price           Decimal?
+  addOn           SlotAddOn[]
```
Comment (kevqiu) [inline prisma/schema.prisma:27]: rename this to `addons`

File: prisma/schema.prisma [hunk 3]
```diff
@@ -24,4 +24,13 @@ model Slot {
...
+}
+
+model SlotAddOn {
```
Comment (kevqiu) [inline prisma/schema.prisma:30]: `SlotAddon`

File: src/slots/index.ts [hunk 4]
```diff
@@ -204,10 +223,34 @@ router.patch('/:id', async (req: Request, res: Response) => {
     if (service !== undefined) data.service = service;
     if (instagramHandle !== undefined) data.instagramHandle = instagramHandle;
     if (price !== undefined) data.price = price;
-
+    if (addOn !== undefined) data.addOn = addOn;
+
+    // if addon create slotaddon loop
+    if (addOn.length > 0) {
+      addOn.forEach(async (element: any) => {
```
Comment (kevqiu) [inline src/slots/index.ts:230]: change `element` to `addon`

File: src/slots/index.ts [hunk 5]
```diff
@@ -185,8 +188,24 @@ router.patch('/:id', async (req: Request, res: Response) => {
       service,
       instagramHandle,
       price,
+      addOn,
```
Comment (kevqiu) [inline src/slots/index.ts:191]: request body should take in value of `addons`

File: src/slots/index.ts [hunk 6]
```diff
@@ -204,10 +223,34 @@ router.patch('/:id', async (req: Request, res: Response) => {
...
+        });
+      });
+    }
     const updatedSlot = await prisma.slot.update({
       where: { id },
-      data,
+      data: {
+        booked: true,
+        addOn: {
+          connectOrCreate: {
```
Comment (kevqiu) [inline src/slots/index.ts:245]: i think you can just call connect on the newly created ones. get the ids of the newly created `addon`s and call connect here
```
connect: [{ id: 8 }, { id: 9 }, { id: 10 }]
```

File: src/slots/index.ts [hunk 7]
```diff
@@ -204,10 +223,34 @@ router.patch('/:id', async (req: Request, res: Response) => {
...
+            where: { id },
+            create: {
```
Comment (kevqiu) [inline src/slots/index.ts:247]: dont do this, this is creating an empty addon for no reason

File: src/slots/index.ts [hunk 8]
```diff
@@ -204,10 +223,34 @@ router.patch('/:id', async (req: Request, res: Response) => {
     if (service !== undefined) data.service = service;
     if (instagramHandle !== undefined) data.instagramHandle = instagramHandle;
     if (price !== undefined) data.price = price;
-
+    if (addOn !== undefined) data.addOn = addOn;
+
+    // if addon create slotaddon loop
```
Comment (kevqiu) [inline src/slots/index.ts:228]: `const addonIds = []`

File: src/slots/index.ts [hunk 9]
```diff
@@ -204,10 +223,34 @@ router.patch('/:id', async (req: Request, res: Response) => {
     if (service !== undefined) data.service = service;
     if (instagramHandle !== undefined) data.instagramHandle = instagramHandle;
     if (price !== undefined) data.price = price;
-
+    if (addOn !== undefined) data.addOn = addOn;
+
+    // if addon create slotaddon loop
+    if (addOn.length > 0) {
+      addOn.forEach(async (element: any) => {
+        await prisma.slotAddOn.create({
```
Comment (kevqiu) [inline src/slots/index.ts:231]: `const createAddon = await prisma...`

File: src/slots/index.ts [hunk 10]
```diff
@@ -204,10 +223,34 @@ router.patch('/:id', async (req: Request, res: Response) => {
...
+    if (addOn.length > 0) {
+      addOn.forEach(async (element: any) => {
+        await prisma.slotAddOn.create({
+          data: {
+            slotId: id,
+            addon: element.addon,
+            price: element.price,
+          },
+        });
+      });
```
Comment (kevqiu) [inline src/slots/index.ts:238]: `addonIds.push({id: createdAddon.id})`

File: src/slots/index.ts (hunk 6 above)
Comment (kevqiu) [reply src/slots/index.ts:245]: `connect:  addonIds`

**Pull request merged**